packages/block-editor/src/components/floating-toolbar/index.native.js

```diff
@@ -18,30 +25,88 @@ import styles from './styles.scss';
 import NavigateUpSVG from './nav-up-icon';
 import Breadcrumb from '../block-list/breadcrumb.native';
 
+const EASE_IN_DURATION = 250;
+const EASE_OUT_DURATION = 80;
+const TRANSLATION_RANGE = 8;
+
+const opacity = new Animated.Value( 0 );
+
 const FloatingToolbar = ( {
 	selectedClientId,
 	parentId,
 	showFloatingToolbar,
 	onNavigateUp,
 	isRTL,
-} ) =>
-	!! showFloatingToolbar && (
-		<TouchableWithoutFeedback accessible={ false }>
-			<View style={ styles.floatingToolbar }>
-				{ !! parentId && (
-					<Toolbar passedStyle={ styles.toolbar }>
-						<ToolbarButton
-							title={ __( 'Navigate Up' ) }
-							onClick={ () => onNavigateUp( parentId ) }
-							icon={ <NavigateUpSVG isRTL={ isRTL } /> }
-						/>
-						<View style={ styles.pipe } />
-					</Toolbar>
-				) }
-				<Breadcrumb clientId={ selectedClientId } />
-			</View>
-		</TouchableWithoutFeedback>
+} ) => {
+	// sustain old selection for proper Breacdrumb rendering when exit animation is ongoing
+	const [ previousSelection, setPreviousSelection ] = useState( {} );
+
+	useEffect( () => {
+		Animated.timing( opacity, {
+			toValue: showFloatingToolbar ? 1 : 0,
+			duration: showFloatingToolbar
+				? EASE_IN_DURATION
+				: EASE_OUT_DURATION,
+			easing: Easing.ease,
+			useNativeDriver: true,
+		} ).start();
+	}, [ showFloatingToolbar ] );
+
+	useEffect( () => {
+		if ( showFloatingToolbar )
+			setPreviousSelection( { clientId: selectedClientId, parentId } );
+	}, [ selectedClientId ] );
+
+	const translationRange =
+		( Platform.OS === 'android' ? -1 : 1 ) * TRANSLATION_RANGE;
+
+	const translation = opacity.interpolate( {
+		inputRange: [ 0, 1 ],
+		outputRange: [ translationRange, 0 ],
+	} );
+
+	const animationStyle = {
+		opacity,
+		transform: [ { translateY: translation } ],
+	};
+
+	const {
+		clientId: previousSelectedClientId,
+		parentId: previousSelectedParentId,
+	} = previousSelection;
+
+	const showPrevious = previousSelectedClientId && ! showFloatingToolbar;
+	const breadcrumbClientId = showPrevious
+		? previousSelectedClientId
+		: selectedClientId;
+	const showNavUpButton =
+		!! parentId || ( showPrevious && !! previousSelectedParentId );
+
+	return (
+		!! opacity && (
+			<TouchableWithoutFeedback accessible={ false }>
```

Comment:
Just out of curiosity, do we really need that `TouchableWithoutFeedback`? What would happen if we remove it and leave the `Animated.View` as a wrapper?

Reply:
I think it make sense. One thing to check if a11y will work as expected after. Will try to cover that today

Reply:
I have tested the a11y on both platform with that solution and it works fine so I will apply changes

Reply:
Awesome, thanks :)